pytensor/link/numba/dispatch/elemwise_codegen.py

```diff
@@ -35,15 +35,20 @@ def compute_itershape(
                 with builder.if_then(
                     builder.icmp_unsigned("!=", length, shape[i]), likely=False
                 ):
-                    with builder.if_else(builder.icmp_unsigned("==", length, one)) as (
+                    with builder.if_else(
+                        builder.or_(
```

Comment:
Weird the changes cause a SegmentationFault on the BroadcastTo numba test, but only on python 3.11? I couldn't replicate locally on 3.8 either. https://github.com/pymc-devs/pytensor/actions/runs/5507826611/jobs/10039563156?pr=372

Did I do something obviously wrong @aseyboldt?

Reply:
I don't see anything wrong, I can try locally with py311 and if I can reproduce I can try to look at it in a debugger (with no debugging symbols, but well...)

Reply:
If you can quickly try to reproduce that's already helpful (even if you don't dig down)

Reply:
No luck so far, for me the tests run just fine...

Reply:
It reliably segfaults here. I'll remove the numba changes for now and put the new test as an xfail

Reply:
Does it segfault during the test_BroadcastTo test?

Reply:
But I don't see how it could be a problem in those tests. There is nothing else in the compiled graph.